### PR TITLE
Added typescript support for react-vis

### PR DESCRIPTION
**Pull request by Domino987**
Added Typescript definitions for react-vis ( https://uber.github.io/react-vis/ )

Please fill in this template.

- [ ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/react-vis/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,3868 @@
...
+
+        yBaseValue?: number;
+
+        yDistance?: number;
+
+        yDomain?: Array<number>;
+
+        yRange?: Array<number>;
+
+        yType?: ('linear' | 'ordinal' | 'category' | 'time' | 'time-utc' | 'log' | 'literal');
```
Comment (andy-ms) [inline types/react-vis/index.d.ts]: A type alias for `('linear' | 'ordinal' | 'category' | 'time' | 'time-utc' | 'log' | 'literal')` would be useful.

File: types/react-vis/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,3868 @@
...
+
+        onDoubleClick?: (datapoint: number, info: infoObject) => void;
+
+        onMouseDown?: (datapoint: number, info: infoObject) => void;
+
+        onMouseEnter?: (datapoint: number, info: infoObject) => void;
+
+        onMouseLeave?: (datapoint: number, info: infoObject) => void;
+
+        onMouseMove?: (datapoint: number, info: infoObject) => void;
```
Comment (andy-ms) [inline types/react-vis/index.d.ts]: A type alias for `(datapoint: number, info: infoObject) => void;` would also be useful.

File: types/react-vis/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,3868 @@
...
+
+        height?: number;
+
+        margin?: number | Object;
+
+        style?: Object;
+
+        tickFormat?: (tickValue: Date | number) => string | number;
+
+        width
```
Comment (andy-ms) [inline types/react-vis/index.d.ts]: This doesn't seem to be used anywhere?

**Review by ghost — CHANGES_REQUESTED** [review]
Needs a file `types/react-vis/react-vis-tests.ts` showing example usage of the library.
I can also tell there are some lint errors, so be sure to run `npm run lint react-vis`.

File: types/react-vis/index.d.ts (hunk 3 above)
Comment (andy-ms) [reply types/react-vis/index.d.ts]: I added this comment on line 3856 (`export type PropsmakeHeightFlexible`) and that's how it shows in the files view, a github bug must be making it show on line 2602 here.

**Pull request closed without merging**